src/UnitVector3d.cc

```diff
@@ -0,0 +1,85 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// \file
+/// \brief This file contains the Vector3d class implementation.
+
+#include "UnitVector3d.h"
+
+#include <cstdio>
+#include <ostream>
+
+
+namespace lsst {
+namespace sg {
+
+UnitVector3d UnitVector3d::orthogonalTo(Vector3d const & v) {
+    Vector3d n;
+    // Pick a vector not equal to v that isn't nearly parallel to it.
+    if (std::fabs(v.x()) > std::fabs(v.y())) {
+        n.y() = 1.0;
+    } else {
+        n.x() = 1.0;
+    }
+    return UnitVector3d(v.cross(n));
+}
+
+UnitVector3d UnitVector3d::orthogonalTo(Vector3d const & v1,
+                                        Vector3d const & v2)
+{
+    Vector3d n = (v2 + v1).cross(v2 - v1);
+    if (n == Vector3d()) {
```

Comment:
How about having a Vector3d::isZero() predicate to use here instead?  It might be more readable, and at least conceptually avoid redundant construction of zero vectors in these comparisons?